s3fs/core.py

```diff
@@ -215,6 +226,7 @@ def info(self, path):
 
     def walk(self, path):
         """ Return all entries below path """
+        path = path.lstrip('s3://')
```

Comment:
OK, thank you. 
I'm not happy how often such path mangling happens through the code, also in the path_split function.